Tool/Sources/XPCShared/XPCServiceProtocol.swift

```diff
@@ -35,6 +35,10 @@ public protocol XPCServiceProtocol {
         editorContent: Data,
         withReply reply: @escaping (Data?, Error?) -> Void
     )
+    func openChat(
```

Comment:
@intitni one question. I want to show this command only if codeium is setup to open chat. Additionally, I want to be able to call this directly. Right now it goes through the XPCService to get API keys etc, do you know of a better way

Reply:
Can you explain what do you mean by service target?

Reply:
> I would suggestion adding a picker in the chat settings to allow users to select what the open chat command will do. And handle it in `func chatWithSelection(editor: EditorContent)` in WindowBaseCommandHandler.swift.
> 
> It's just not recommended to add dynamic items to the Xcode menu because it can't be refreshed until Xcode is restarted.

I mean this.